metals/src/main/scala/scala/meta/internal/metals/FileDecoderProvider.scala

```diff
@@ -190,12 +190,21 @@ final class FileDecoderProvider(
     new URI("jar", decoded.stripPrefix("jar:"), null)
   }
 
-  private def decodeJar(uri: URI): DecoderResponse = {
+  private def decodeJar(uri: URI, original: String): DecoderResponse = {
     Try {
       val path = uri.toAbsolutePath
       FileIO.slurp(path, StandardCharsets.UTF_8)
     } match {
-      case Failure(exception) => DecoderResponse.failed(uri, exception)
+      case Failure(_) =>
```

Comment:
Do we know what is thrown here if it fails? Is this just a `FileNotFound` exception? If so it _might_ be nicer to instead do like a:

```
  private def decodeJar(
      uri: URI,
      fallback: Option[String] = None,
  ): DecoderResponse = {
    Try {
      val path = uri.toAbsolutePath
      FileIO.slurp(path, StandardCharsets.UTF_8)
    } match {
      case Failure(_: FileNotFoundException) if fallback.nonEmpty =>
        decodeJar(new URI(fallback.get))
      case Failure(exception) =>
        scribe.warn(s"Unable to decode: $fallback", exception)
        DecoderResponse.failed(uri, exception)
      case Success(value) => DecoderResponse.success(uri, value)
    }
  }
```
But take it or leave it.

Reply:
It can be either FileNotFoundException or FilesystemNotFound, so I might use that, thanks!

Reply:
Ok, so I ended up not doing this since the fallback uses the toAbsolutePath on String and not `new URI(fallback.get)`. String.toAbsolutePath  could work in all situations, but since this is rather fragile I will keep the old behaviour and fallback to it.

I changed it though, so it's much shorter while doing the same thing.